lcls2_producers/smd_producer.py

```diff
@@ -344,134 +378,151 @@ def get_sd_file(write_dir, exp, hutch):
     print(f"MPI size: {size}")
     print(f"PS_EB_NODES={os.environ.get('PS_EB_NODES')}")
     print(f"PS_SRV_NODES={os.environ.get('PS_SRV_NODES')}")
-    print(f"PS_SMD_N_EVENTS={os.environ.get('PS_SMD_N_EVENTS')}") # defaults to 1000
+    print(f"PS_SMD_N_EVENTS={os.environ.get('PS_SMD_N_EVENTS')}")  # defaults to 1000
     print(f"DS batchsize: {ds.batch_size}")
     print("#### END DATASOURCE AND PSANA ENV VAR INFO ####\n")
 
 thisrun = next(ds.runs())
 
 # Generate smalldata object
 if ds.unique_user_rank():
-    print('Opening the h5file %s, gathering at %d'%(h5_f_name,args.gather_interval))
-small_data = ds.smalldata(filename=h5_f_name, batch_size=args.gather_interval)
+    print("Opening the h5file %s, gathering at %d" % (h5_f_name, args.gather_interval))
+if args.psplot_live_mode:
+    logger.info("Setting up psplot_live plots.")
+    psplot_configs = config.get_psplot_configs(int(run))
+
+    psplot_callbacks = psplot.PsplotCallbacks()
+    for key, item in psplot_configs.items():
+        plot_type = item.pop("plot_type")
+        psplot_callbacks.add_callback(plot_type(item), name=key)
+    small_data = ds.smalldata(
+        filename=None, batch_size=args.gather_interval, callbacks=[psplot_callbacks.run]
+    )
+else:
+    small_data = ds.smalldata(filename=h5_f_name, batch_size=args.gather_interval)
 if ds.unique_user_rank():
     print(f"rank: {rank}")
-    print('smalldata file has been successfully created.')
-
+    print("smalldata file has been successfully created.")
```

Comment:
These are the important lines for the psplot live implementation (black has done a lot on this file...)